src/Npgsql/Internal/TypeHandlers/ArrayHandler.cs

```diff
@@ -152,7 +152,7 @@ protected async ValueTask<Array> ReadArray<TRequestedElement>(NpgsqlReadBuffer b
 
                 // TODO: Overly complicated/inefficient...
                 indices[dimensions - 1]++;
-                for (var dim = dimensions - 1; dim >= 0; dim--)
+                for (var dim = indices.Length - 1; dim >= 0; dim--)
```

Comment:
For this `for`-loop bound checks won't be removed for current and previous JIT.
In L160 `dim == 0` is checked, is the loop-condition here necessary. Could this be a `while`-loop instead?

Note: I saw the TODO comment above, so maybe my comment here is to be ignored.

Reply:
GH doesn't allow me to comment on L163. This could be 
```c#
for (var j = dim; j < indices.Length; j++)
```
to remove the bound checks there (at least for .NET 6).